### PR TITLE
kubeadm: organize phases cmd

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Reorganize cmd for phases, by separating init phases and join phases

**Which issue(s) this PR fixes**:
none

**Special notes for your reviewer**:
While splitting the control-plane-prepare phase, I simplified the manifest subphase in order to avoid the third level of nested phases (if we want to add this, we should do this consistently for certs, kubeconfig and manifests)

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cluster-lifecycle
/area kubeadm
/priority backlog
@kubernetes/sig-cluster-lifecycle-pr-reviews
/assign @neolit123 
/assign @ereslibre 
@MalloZup

### CODE REVIEW

**Review by yagonobre — COMMENTED** [review]
Thanks @fabriziopandini 
/lgtm

File: cmd/kubeadm/app/cmd/phases/join/controlplane.go [hunk 1]
```diff
@@ -182,59 +85,37 @@ func newControlPlanePrepareKubeconfigSubphase() workflow.Phase {
...
-func newControlPlanePrepareSubphase(component string) workflow.Phase {
-	return workflow.Phase{
-		Name:         controlPlanePhaseProperties[component].name,
-		Short:        controlPlanePhaseProperties[component].short,
-		Run:          runControlPlanePrepareJoinSubphase(component),
-		InheritFlags: getControlPlanePreparePhaseFlags(),
+		Name:         "manifests",
+		Short:        "Generates the manifests for the new control plane components",
+		Run:          runControlPlaneSubphase,
+		InheritFlags: getControlPlanePreparePhaseFlags(), //NB. eventually in future we would like to break down this in sub phases for each component
```
Comment (ereslibre) [inline cmd/kubeadm/app/cmd/phases/join/controlplane.go]: Why lose the subphases here and add the comment to add the subphases in the future?

**Comment by fabriziopandini** [comment]
Removing WIP as per comments/feedbacks on slack

**Comment by fabriziopandini** [comment]
@chuckha here it is the thread for better visibility, I'm going to add a comment to 1375

@fabriziopandini 
> Hi all, just sent https://github.com/kubernetes/kubernetes/pull/73725
> It re-organize phases, by splitting init phases and join phases
> + I simplified the manifest subphase in order to avoid the third level of nested phases (if we want to 
> add this, we should do this consistently for certs, kubeconfig and manifests)
> it is still a WIP because those are changes not agreed before
> let me know if you are fine with those changes or not

@neolit123 
> i think this is great 👍 

@ereslibre 
> 1) I think subdirectories as you did is a better approach than sibling files as I first started, 
> 2) I think it makes sense to remove the subphases for manifests because we are not coherent
> with subphases in certs and kubeconfig so yes, :+1: from me, the organization is clearly better

**Pull request merged**